app/src/main/java/tw/fatminmin/xposed/minminguard/blocker/ApiBlocking.java

```diff
@@ -101,14 +101,55 @@ protected void beforeHookedMethod(MethodHookParam param) throws Throwable {
         return true;
     }
 
-    public static boolean blockAdFunction(final String packageName, final String ad, final String adFunc, final Class<?> parameter, final XC_LoadPackage.LoadPackageParam lpparam, final boolean removeAd) {
+    public static boolean blockAdFunction(final String packageName, final String ad, final String adFunc, final Object parameter, final XC_LoadPackage.LoadPackageParam lpparam, final boolean removeAd) {
```

Comment:
I think we should be able to use variable arguments to merge all the overloaded `blockAdFunction` concisely.

Reply:
https://stackoverflow.com/questions/2925153/can-i-pass-an-array-as-arguments-to-a-method-with-variable-arguments-in-java

Reply:
Yes, I know of that. But then you have to include the callback as a vararg. If you dont, Xposed will throw an error, as what you are actually doing is passing it an object[] and then the callback, when the callback should be passed in the object[]. [Here](https://github.com/rovo89/XposedBridge/issues/189) is an explination by rovo89

Reply:
Thanks for explaination :+1: